tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp

```diff
@@ -88,21 +88,40 @@ std::shared_ptr<trajopt::ProblemConstructionInfo> TrajOptPlannerDefaultConfig::g
   pci.basic_info.use_time = false;
   pci.basic_info.convex_solver = optimizer;
 
+  // Get kinematics information
+  tesseract_environment::Environment::ConstPtr env = tesseract->getEnvironmentConst();
+  tesseract_kinematics::ForwardKinematics::ConstPtr kin =
+      tesseract->getFwdKinematicsManagerConst()->getFwdKinematicSolver(manipulator);
+  tesseract_environment::AdjacencyMap map(
+      env->getSceneGraph(), kin->getActiveLinkNames(), env->getCurrentState()->transforms);
+  std::vector<std::string> adjacency_links = map.getActiveLinkNames();
+
   // Populate Init Info
   pci.init_info.type = init_type;
   if (init_type == trajopt::InitInfo::GIVEN_TRAJ)
+  {
     pci.init_info.data = seed_trajectory;
 
+    // Add check to make sure if starts with joint waypoint that the seed trajectory also starts at this waypoint
+    // If it does not start this causes issues in trajopt and it will never converge.
+    if (isJointWaypointType(target_waypoints.front()->getType()))
+    {
+      const auto jwp = std::static_pointer_cast<JointWaypoint>(target_waypoints.front());
+      const Eigen::VectorXd position = jwp->getPositions(kin->getJointNames());
+      for (int i = 0; i < static_cast<int>(kin->numJoints()); ++i)
+      {
+        if (std::abs(position[i] - seed_trajectory(0, i)) > static_cast<double>(std::numeric_limits<float>::epsilon()))
+        {
+          CONSOLE_BRIDGE_logError("Seed trajectory start position does not match starting joint waypoint position!");
```

Comment:
I'd prefer the logged error indicated the joint that failed. Then you'd have to go through every one but return false if one or more failed.

Reply:
Fixing it now.